server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1143,37 +1183,36 @@ private SegmentIdWithShardSpec createNewSegment(
                                  : PartitionIds.ROOT_GEN_START_PARTITION_ID;
 
       String version = newSegmentVersion == null ? existingVersion : newSegmentVersion;
-      return new SegmentIdWithShardSpec(
+      SegmentIdWithShardSpec segmentId = new SegmentIdWithShardSpec(
           dataSource,
           interval,
           version,
           partialShardSpec.complete(jsonMapper, newPartitionId, 0)
       );
+      return SegmentAllocateResult.success(segmentId);
     } else if (!overallMaxId.getInterval().equals(interval)
                || overallMaxId.getVersion().compareTo(existingVersion) > 0) {
-      log.warn(
-          "Cannot allocate new segment for dataSource[%s], interval[%s], existingVersion[%s]: conflicting segment[%s].",
-          dataSource,
-          interval,
-          existingVersion,
-          overallMaxId
+      return SegmentAllocateResult.failure(
+          "Segment[%s] in conflicting interval[%s] has higher version[%s].",
```

Comment:
This error message is inaccurate. It's also possible that the interval corresponding to overallMaxId isn't the same as the request's interval.

Reply:
Thanks for catching this, I will break it up.